Implementation of issue 8: block all/unblock all from command line

jappeace opened the pull request:
Hi

I had some time on my hands and decided to try and implement this (I'm new at Rust, and it seemed easy enough). The options are presented with -h and if no options are given then it defaults back to the menu version.

If you like the changes please merge them. If you don't, I would like to receive some critique.

Cheers

jappeace committed:
vim swap file

jappeace committed:
issue #8 -b to block all now works

jappeace committed:
fixed #8 can now also unblock trough -u

jappeace committed:
extended the readme with new features of #8

jappeace committed:
formatting of readme

jappeace committed:
added extra clearification for help

jappeace committed:
help message no longer claims that you need to insert a file

jappeace committed:
better grammar

cgag commented:
@jappeace Awesome, thanks, I love pull requests. I'm think I'm still against the -u option, but I'm all for adding the "block all" option as @theRealPadster mentioned.  I just looked the code, I'll do another read through before merging, but the things that jumped out at me are:

 1) The "// TODO: pattern matching" comment. I definitely agree pattern matching would be nice there.  I'm not sure the ideal way to go about it.  It looks like the matches object is a vector of strings, since there's only one argument, I think just pattern matching on the first element of the vector would be good.

2) Since block/unblock all return a new state, could we just do something like `state = block_all(state)`, instead of individually setting fields on the existing state?

I'm also never thrilled about returning a boolean vs some sort of status value (like I did handle key), as is done with read_args, but that's not a big deal.

Let me know if you plan on making any of those changes and I'll wait to merge it.  Otherwise I'm happy as is and I'll make any tweaks when I have time, none of the things I mentioned are showstoppers.  Thanks a bunch for the PR.

jappeace commented:
Yes I can implement the changes, 1 especially (coming from Scala I love the pattern matching feature), and for 2 I tried doing that, but its not allowed (since you're borrowing that reference), normally in C++ you would just dereference and overwrite the value, but I have to agree with the compiler that its a weird thing to do to drastically change the state of an input variable (or completely replace it), so I was thinking about letting a state and a boolean be returned or even let the function create the state. I probably should also change the name of the function or split it up or something if I do that.

I was also thinking about making the pass phrase longer for -u as discussed in #8. So I was thinking about having 2 extra words for -u, 1 word for the selection "mental activity" and one for the context switch (assuming here it takes some time to switch activities).

So yes just decline this pull request, I'll implement the changes (still have some extra time on my hands) and I'll improve it and re-do the pull request.

jappeace committed:
this functionally works -u has now 2 more words to type

jappeace committed:
allows the compile flag for commandline unblock

jappeace committed:
added comment

jappeace committed:
pattern matching isn't possible in this case, because we use a function
call (with varying arguments) instead of an assigned value. Therefore I
deleted the comment

jappeace committed:
this does some pattern matching, but I don't think its a lot better then
the previous solution

jappeace committed:
deleted the patern matching solution as it was a trivial improvement
